filament/backend/src/metal/MetalDriver.mm

```diff
@@ -706,16 +706,26 @@
     scheduleDestroy(std::move(data));
 }
 
-void MetalDriver::setExternalIndexBuffer(Handle<HwIndexBuffer> ibh, void* externalBuffer) {
+void MetalDriver::setupExternalResource(intptr_t externalResource) {
+    CFTypeRef bufferRef = (CFTypeRef) externalResource;
+    CFRetain(bufferRef);
```

Comment:
Better to add a null check, `CFRetain()` and `CFRelease()` are crashing on null - contrary to their specializations (e.g. `CVPixelBufferRetain()`)

Reply:
Like this? a619fabff10e5da2ebff62a7b4e3edd56bfad1ea

Reply:
[In my latest commit](https://github.com/shapr3d/filament/pull/46/commits/fbd7516aa11aa15ef6a7313cd86941a68660960a#diff-d460ae873f0517c57e6465122e338cbab93525afdc4605088ba77962a3a79be1R715) the logic is the same for `NULL` safety like in the previously mentioned commit, but I reverted the `CFTypeRef` casting to the version what was in `setupExternalImage` originally.